### ISSUE TITLE
Implement serde traits for `Weight`

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#3816]
For types that do not have a specified serialisation format we provide explicit functions so users are forced to make a decision when serializing. We do this for

- `Amount`
- `SignedAmount`
- `FeeRate` (in #3666)

Since `Weight` has a unit (weight units), has no specified serialisation format, and could be confused with virtual bytes perhaps we should do the same for `Weight`?

**Comment by apoelstra** [comment]
No, `Weight` is literally named after the unit it represents. It is an integer number of weight units. We should just serialize it as an integer. There is no other way that makes sense.

Anybody who legitimately confuses weight with vb (not just making typos that hopefully the compiler will catch) has no hope of writing correct code anyway and we don't need to worry about them.

**Comment by tcharding** [comment]
I added a serde label because there is a to of serde work that needs doing soon.

**Comment by jrakibi** [comment]
Will work on this

**Comment by tcharding** [comment]
Ooph, this issue has been NACK'ed - it should not be open still. My bad.
